Sources/Bugsnag/BugsnagMiddleware.swift

```diff
@@ -2,42 +2,44 @@ import Vapor
 import HTTP
 
 public final class BugsnagMiddleware: Middleware {
-    
+
     let drop: Droplet
-    let configuration: Configuration
-    let connectionManager: ConnectionMananger
+    let configuration: ConfigurationType
+    let connectionManager: ConnectionManagerType
+
     public init(drop: Droplet) throws {
         self.drop = drop
         self.configuration = try Configuration(drop: drop)
         self.connectionManager = ConnectionMananger(drop: drop, config: configuration)
     }
 
+    internal init(connectionManager: ConnectionManagerType) {
+        self.drop = connectionManager.drop
+        self.configuration = connectionManager.config
+        self.connectionManager = connectionManager
+    }
+
+
+    // MARK: - Middleware
+
     public func respond(to request: Request, chainingTo next: Responder) throws -> Response {
-        
         do {
             return try next.respond(to: request)
-        } catch Abort.badRequest {
-            try report(status: .badRequest, message: "Bad request", request: request)
-            throw Abort.badRequest
-        } catch Abort.serverError {
-            try report(status: .internalServerError, message: "Server error", request: request)
-            throw Abort.serverError
-        } catch Abort.notFound {
-            try report(status: .notFound, message: "Not found", request: request)
-            throw Abort.notFound
-        } catch Abort.custom(let status, let message) {
-            try report(status: status, message: message, request: request)
-            throw Abort.custom(status: status, message: message)
+        } catch let error as AbortError {
+            if error.metadata?["report"]?.bool ?? true {
+                try report(status: error.status, message: error.message, metadata: error.metadata, request: request)
+            }
+            throw error
+        } catch {
+            try report(status: .internalServerError, message: Status.internalServerError.reasonPhrase, metadata: nil, request: request)
+            throw error
         }
-        // This can be incommented when Vapor updates
-        /*catch Abort.customWithCode(let status, let message, let code) {
-            try report(status: status, message: message, request: request)
-            throw Abort.customWithCode(status: status, message: message, code: code)
-        }*/
     }
-    
-    public func report(status: Status, message: String, request: Request) throws {
-        _ = try connectionManager.post(status: status, message: message, request: request)
+
```

Comment:
Consider having a global func to report errors.

in case you want something like this
```swift
do {
    try something 
} catch {
bugsnagReport(error)
}